pkgs/applications/science/electronics/kicad/base.nix

```diff
@@ -151,15 +153,19 @@ stdenv.mkDerivation rec {
     curl
     openssl
     boost
-    swig
+    swig4
     python
   ]
+  ++ optional (!stable) unixODBC
   ++ optional (withScripting) wxPython
   ++ optional (withNgspice) libngspice
   ++ optional (withOCC) opencascade-occt
   ++ optional (debug) valgrind
   ;
 
+  # started becoming necessary halfway into 2022, not sure what changed to break a test...
+  preInstallCheck = optionals (withNgspice) [ "export LD_LIBRARY_PATH=${libngspice}/lib" ];
```

Comment:
This probably means that this library is somehow needed at build time. I'm not sure, but maybe adding the library also to `nativeBuildInputs` would have the same effect (not tested).

Reply:
fair point, i didn't actually test the spice simulation part, as i'm not familiar with that...

Reply:
adding the library to nativeBuildInputs instead setting it for the installCheck phase results in the same failure

going to give spice simulation a go next

Reply:
simulation seems to work

i should probably open an issue about it not finding the library...

Reply:
https://gitlab.com/kicad/code/kicad/-/issues/12491